lib/openmic.rb

```diff
@@ -0,0 +1,50 @@
+class OpenMic
+  attr_reader :location, :date, :performers
+
+  def initialize(data)
+       @location = data[:location]
+       @date = data[:date]
+       @performers = []
+  end
+
+  def welcome(user)
+       @performers << user
+  end
+
+  def repeated_jokes?
```

Comment:
A simpler way to think about it - If you get a list of all jokes known by performers, you can compare a unique list of those jokes against the full list to see if there are any duplicates (this has the added benefit of testing uniqueness of jokes even within a single comedian) It could look something like this:

```ruby
def repeated_jokes?
  all_jokes = @performers.map do |performer|
    performer.jokes
  end.flatten

  all_jokes.uniq < all_jokes
end
```